File: Tools/PHP/access_bd.php
<?php
class Database
{
    private $host = '10.40.42.13';
    private $port = '80';
    private $dbname = 'nombre_de_tu_base';
    private $username = 'tu_usuario';
    private $password = 'tu_contraseña';
    private $pdo;

    // Constructor para establecer la conexión
    public function __construct()
    {
        try {
            $dsn = "sqlsrv:Server={$this->host},{$this->port};Database={$this->dbname}";
            $this->pdo = new PDO($dsn, $this->username, $this->password);
            // Configuraciones opcionales de PDO
            $this->pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
            $this->pdo->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
        } catch (PDOException $e) {
            die("Error en la conexión: " . $e->getMessage());
        }
    }

    // Método para ejecutar una consulta
    public function query($sql, $params = [])
    {
        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute($params);
            return $stmt->fetchAll();
        } catch (PDOException $e) {
            die("Error en la consulta: " . $e->getMessage());
        }
    }

    // Método para cerrar la conexión
    public function closeConnection()
    {
        $this->pdo = null;
    }
}

// Ejemplo de uso:
$db = new Database();
$resultado = $db->query("SELECT * FROM tabla");
print_r($resultado);
$db->closeConnection();

File: Tools/Python/access_bd.py
import pyodbc

class Database:
    def __init__(self, server='10.40.42.13', port='80', database='nombre_de_tu_base', username='tu_usuario', password='tu_contraseña'):
        try:
            self.connection = pyodbc.connect(
                f'DRIVER={{ODBC Driver 17 for SQL Server}};'
                f'SERVER={server},{port};'
                f'DATABASE={database};'
                f'UID={username};'
                f'PWD={password}'
            )
            self.cursor = self.connection.cursor()
        except pyodbc.Error as e:
            print("Error en la conexión: ", e)
    
    # Método para ejecutar consultas
    def query(self, sql, params=()):
        try:
            self.cursor.execute(sql, params)
            return self.cursor.fetchall()
        except pyodbc.Error as e:
            print("Error en la consulta: ", e)
            return None
    
    # Método para cerrar la conexión
    def close_connection(self):
        self.cursor.close()
        self.connection.close()

# Ejemplo de uso:
db = Database()
resultado = db.query("SELECT * FROM tabla")
for fila in resultado:
    print(fila)
db.close_connection()
